deployment/roles/toy.database/tasks/setup-db.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Make sure template_postgis exists
+  postgresql_db: db=template_postgis state=present encoding=UTF-8 template=template0
+  become_user: postgres
+  register: createdb_template_postgis
+
+- name: Spatially enable the template database
+  command: psql -d template_postgis -c "CREATE EXTENSION IF NOT EXISTS postgis;"
+  when: createdb_template_postgis.changed
+  become_user: postgres
+
+- name: Create admin user
+  command: psql -c "create user admin with password 'fakepassword';"
```

Comment:
This (and many of the tasks in this file) will fail if the user tries to run provisioning twice, because they don't have a way to detect if the task they perform has already been done. I think in most of these cases, there is an Ansible plugin that provides equivalent functionality, which can help with rerunning provisioning. For user creation, you can use https://docs.ansible.com/ansible/latest/collections/community/general/postgresql_user_module.html

For the tasks for which you can't find a preexisting Ansible plugin, I think you should be able to use the `register` / `.changed` pattern from above to avoid running things twice. You could potentially also use Handlers: https://docs.ansible.com/ansible/latest/user_guide/playbooks_handlers.html

Reply:
Hmmm, I'm a little bit confused with regard to this comment. Is [this implementation](https://github.com/azavea/cac-tripplanner/blob/681fa2dc95ea8f6a330844babed122254db7ab5e/deployment/ansible/roles/cac-tripplanner.database/tasks/main.yml#L35) in cac-tripplanner that you linked in one of the other comments here following the pattern that you are describing here?

Reply:
Yep, exactly! Sorry, I should have provided an example along with the comment, but that's exactly the kind of thing I was thinking of.

Reply:
Gotcha. So when the Ansible `postgresql_db` and `postgresql_user` modules, the issues with rerunning provisioning are mitigated?

Reply:
Exactly -- all Ansible tasks are required to be idempotent, so if they condition that they're seeking to enforce already exists, they'll bail out and leave things unchanged.